Content.Client/Xenoarchaeology/Ui/AnalysisConsoleMenu.xaml.cs

```diff
@@ -15,169 +24,235 @@ namespace Content.Client.Xenoarchaeology.Ui;
+        else if (arti == null)
         {
-            ExtractButton.Disabled = true;
-
-            // CanScan can be false if either there's no analyzer connected or if there's
-            // no entity on the scanner. The `Information` text will always tell the user
-            // of the former case, but in the latter, it'll only show a message if a scan
-            // has never been performed, so add a tooltip to indicate that the artifact
-            // is gone.
-            if (state.AnalyzerConnected)
-            {
-                ExtractButton.ToolTip = Loc.GetString("analysis-console-no-artifact-placed");
-            }
-            else
-            {
-                ExtractButton.ToolTip = null;
-            }
+            NoArtiLabel.Text = Loc.GetString("analysis-console-info-no-artifact");
         }
-        else if (state.PointAmount <= 0)
+        else
         {
-            ExtractButton.Disabled = true;
-            ExtractButton.ToolTip = Loc.GetString("analysis-console-no-points-to-extract");
+            NoArtiLabel.Visible = false;
         }
 
-        if (ExtractButton.Disabled)
-        {
-            ExtractButton.RemoveStyleClass("ButtonColorGreen");
-        }
-        else
+        if (_currentNode == null
+            || arti == null
+            || !_xenoArtifact.TryGetIndex((arti.Value, arti.Value), _currentNode.Value, out _))
         {
-            ExtractButton.AddStyleClass("ButtonColorGreen");
-            ExtractButton.ToolTip = null;
+            SetSelectedNode(null);
         }
     }
-    private void UpdateArtifactIcon(EntityUid? uid)
+
+    public void SetSelectedNode(Entity<XenoArtifactNodeComponent>? node)
     {
-        if (uid == null)
-        {
-            ArtifactDisplay.Visible = false;
+        InfoContainer.Visible = node != null;
+        if (!_artifactAnalyzer.TryGetArtifactFromConsole(_owner, out var artifact))
             return;
-        }
 
-        ArtifactDisplay.Visible = true;
-        ArtifactDisplay.SetEntity(uid);
-    }
+        NoneSelectedLabel.Visible = node == null;
 
-    public void UpdateInformationDisplay(AnalysisConsoleUpdateState state)
-    {
-        var message = new FormattedMessage();
+        if (node == null)
+            return;
+
+        var nodeId = _xenoArtifact.GetNodeId(node.Value);
+        IDValueLabel.SetMarkup(Loc.GetString("analysis-console-info-id-value", ("id", nodeId)));
+
+        // If active, state is 2. else, it is 0 or 1 based on whether it is unlocked, or not.
+        var lockedState = _xenoArtifact.IsNodeActive(artifact.Value, node.Value)
+            ? 2
+            : node.Value.Comp.Locked
+                ? 0
+                : 1;
+        LockedValueLabel.SetMarkup(Loc.GetString("analysis-console-info-locked-value",
+            ("state", lockedState)));
 
-        if (state.Scanning)
+        var percent = (float) node.Value.Comp.Durability / node.Value.Comp.MaxDurability;
+        var color = percent switch
```

Comment:
Actually switch expression was considered one of the 'good ones', standing in line with pattern matching <_< this one i'd better leave as is (but the whole 'lets color number in this method based on some durability node have' is questionable so it could be deleted in near future i feel...)